tel2puml/pv_to_puml.py

```diff
@@ -268,13 +268,108 @@ def pv_streams_to_puml_files(
     "."
-    :type file_directory: `str`
+    :type output_file_directory: `str`
     """
     for job_name, job_event_gen in pv_streams:
-        puml_file_path = os.path.join(file_directory, f"{job_name}.puml")
+        job_name = job_name.replace(" ", "_")
+        puml_file_path = os.path.join(
+            output_file_directory, f"{job_name}.puml"
+        )
         pv_to_puml_file(
             job_event_gen,
             puml_file_path=puml_file_path,
         )
+
+
+def pv_files_to_pv_streams(
+    file_directory: Optional[str] = None,
+    file_list: Optional[list[str]] = None,
+    job_name: str = "default.puml",
+    group_by_job_id: Optional[bool] = False,
+) -> Generator[
+    tuple[str, Generator[Generator[PVEvent, Any, None], Any, None]],
+    Any,
+    None,
+]:
+    """Converts PV job JSON files from a specified directory or a provided file
+    list into a generator of tuples.
+
+    Each tuple consists of the job name and a generator of generators, where
+    each inner generator yields `PVEvent` objects.
+
+    :param file_directory: The directory containing PV job JSON files. Defaults
+    to `None`.
+    :type file_directory: `Optional`[`str`]
+    :param file_list: A list of file paths to PV job JSON files. Defaults
+    to `None`.
+    :type file_list: `Optional`[`list`[`str`]]
+    :param job_name: The name of the job associated with the PV events.
+    Defalts to `default.puml`
+    :type job_name: `str`
+    :param group_by_job_id: Flag to determine whether to group events by
+    their `job_id`. Defaults to `False`
+    :type group_by_job_id: `bool`
+    :return: A generator that yields tuples. Each tuple contains the job name
+    and a generator of generators of `PVEvent` instances.
+    :rtype: `Generator`[`tuple`[`str`,`Generator`[`Generator`[:class:`PVEvent`,
+    `Any`, `None`], `Any`, `None`]],`Any`,`None`]
+    """
+    if file_directory:
+        file_paths: list[str] = [
+            file_directory + "/" + file
+            for file in os.listdir(file_directory)
+            if file.endswith(".json")
+        ]
+    elif file_list:
+        file_paths = file_list
+    else:
+        raise ValueError("file_directory or file_list has to be specified.")
+
+    pv_stream_sequence = pv_job_files_to_event_sequence_streams(file_paths)
+
+    if group_by_job_id:
+        events_by_job_id: dict[str, list[PVEvent]] = {}
+        for pv_stream in pv_stream_sequence:
+            clustered_events = cluster_events_by_job_id(pv_stream)
+            for job_id, events in clustered_events.items():
+                if job_id in events_by_job_id:
+                    events_by_job_id[job_id].extend(events)
+                else:
+                    events_by_job_id[job_id] = events
+
+        pv_stream_sequence = (events for events in events_by_job_id.values())
+
+    pv_stream_gen = event_sequence_to_gen(pv_stream_sequence)
+
+    yield (job_name, pv_stream_gen)
+
+
+def event_sequence_to_gen(
```

Comment:
so we shouldn't need the two functions below. We can just use `pv_stream_sequence` which is  `Generator[list[PVEvent], Any, None]` as long as we change the typing of `pv_streams_to_puml_files` in pv_to_puml.py replacing `Generator` with `Iterable` which is more general and should work